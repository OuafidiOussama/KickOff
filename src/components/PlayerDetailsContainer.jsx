import { View, Text, Image, StyleSheet, ScrollView } from "react-native";
import React from "react";
import { useSelector } from "react-redux";

export default function PlayerDetailsContainer({ playerId }) {
  const player = useSelector((state) =>
    state.Players.players.find((player) => player.id === playerId)
  );
  return (
    <ScrollView>
      <View style={styles.container}>
        <View style={styles.imageBack}>
          <Image
            source={{
              uri: `${player.player_picture}`,
            }}
            style={styles.image}
          />
          <Text style={styles.playerName}>{player.player_name}</Text>
        </View>

        <View style={styles.detail}>
          <Image
            source={{ uri: `${player.team_picture}` }}
            style={styles.teamPicture}
          />
          <View style={styles.info}>
            <Text style={styles.position}>Position</Text>
            <Text style={styles.infoText}> {player.position_short_name}</Text>
          </View>

          <View style={styles.info}>
            <Text style={styles.position}>Age</Text>
            <Text style={styles.infoText}> {player.age}</Text>
          </View>

          <View style={styles.info}>
            <Text style={styles.position}>National Team</Text>
            <Text style={styles.infoText}> {player.country_name}</Text>
          </View>

          <View style={styles.info}>
            <Text style={styles.position}>Team</Text>
            <Text style={styles.infoText}> {player.team_name}</Text>
          </View>

          <View style={styles.info}>
            <Text style={styles.position}>Skill</Text>
            <Text style={styles.infoText}>{player.sci_skill_smg}</Text>
          </View>
          <View style={styles.info}>
            <Text style={styles.position}>Value</Text>
            <Text style={styles.infoText}>{player.estimated_value}</Text>
          </View>
        </View>
      </View>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    height: "100%",
    display: "flex",
    flexDirection: "column",
    gap: 40,
  },
  imageBack: {
    display: "flex",
    alignItems: "center",
    backgroundColor: "#5b8db0",
    padding: 30,
    width: "100%",
    borderBottomEndRadius: 40,
    borderBottomStartRadius: 40,
    shadowColor: "black",
    shadowOffset: {
      width: 0,
      height: 12,
    },
    shadowOpacity: 0.5,
  },
  teamPicture: {
    width: "100%",
    height: "120%",
    position: "absolute",
    opacity: 0.2,
  },
  image: {
    width: 150,
    height: 150,
    marginBottom: 10,
  },
  playerName: {
    fontSize: 20,
    fontWeight: "bold",
    color: "white",
  },
  position: {
    fontSize: 18,
    fontWeight: "bold",
    color: "white",
    textAlign: "center",
    marginBottom: 10,
  },

  infoText: {
    textAlign: "center",
    color: "black",
  },

  detail: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    alignItems: "center",
    gap: 20,
  },
  info: {
    width: "40%",
    marginTop: 20,
    backgroundColor: "#5b8db0",
    padding: 20,
    borderRadius: 40,
    shadowColor: "black",
    shadowOffset: {
      width: 4,
      height: 12,
    },
    shadowOpacity: 0.6,
  },
});
