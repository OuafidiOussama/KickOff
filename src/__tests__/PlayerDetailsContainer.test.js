import React from "react";
import renderer from "react-test-renderer";
import { ScrollView, View, Text, Image } from "react-native";
import { useSelector } from "react-redux";
import PlayerDetailsContainer from "../components/PlayerDetailsContainer";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
}));

describe("<PlayerDetailsContainer />", () => {

  it("renders correctly when player data is available", () => {
    const mockPlayer = {
      id: "1",
      player_picture: "https://static.footballtransfers.com/resources/players/506100.png",
      player_name: "Messi",
      team_picture: "https://static.footballtransfers.com/resources/teams/48.png",
      position_short_name: "FW",
      age: 25,
      country_name: "Morroco",
      team_name: "Team",
      sci_skill_smg: "Excellent",
      estimated_value: "$10M",
    };

    useSelector.mockImplementation((selector) =>
      selector({
        Players: { players: [mockPlayer] },
      })
    );

    const tree = renderer
      .create(<PlayerDetailsContainer playerId="1" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders loading text when player data is not available", () => {
    useSelector.mockReturnValueOnce({
      Players: { players: [] },
    });

    const tree = renderer
      .create(<PlayerDetailsContainer playerId="1" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
