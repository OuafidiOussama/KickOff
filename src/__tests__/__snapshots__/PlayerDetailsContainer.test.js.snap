// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerDetailsContainer /> renders correctly when player data is available 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "gap": 40,
          "height": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#5b8db0",
            "borderBottomEndRadius": 40,
            "borderBottomStartRadius": 40,
            "display": "flex",
            "padding": 30,
            "shadowColor": "black",
            "shadowOffset": {
              "height": 12,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "width": "100%",
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://static.footballtransfers.com/resources/players/506100.png",
            }
          }
          style={
            {
              "height": 150,
              "marginBottom": 10,
              "width": 150,
            }
          }
        />
        <Text
          style={
            {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Messi
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "gap": 20,
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://static.footballtransfers.com/resources/teams/48.png",
            }
          }
          style={
            {
              "height": "120%",
              "opacity": 0.2,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Position
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
             
            FW
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Age
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
             
            25
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            National Team
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
             
            Morroco
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Team
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
             
            Team
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Skill
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
            Excellent
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Value
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
            $10M
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<PlayerDetailsContainer /> renders loading text when player data is not available 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "gap": 40,
          "height": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#5b8db0",
            "borderBottomEndRadius": 40,
            "borderBottomStartRadius": 40,
            "display": "flex",
            "padding": 30,
            "shadowColor": "black",
            "shadowOffset": {
              "height": 12,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "width": "100%",
          }
        }
      >
        <Image
          source={
            {
              "uri": "undefined",
            }
          }
          style={
            {
              "height": 150,
              "marginBottom": 10,
              "width": 150,
            }
          }
        />
        <Text
          style={
            {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "gap": 20,
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            {
              "uri": "undefined",
            }
          }
          style={
            {
              "height": "120%",
              "opacity": 0.2,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Position
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
             
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Age
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
             
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            National Team
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
             
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Team
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          >
             
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Skill
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "#5b8db0",
              "borderRadius": 40,
              "marginTop": 20,
              "padding": 20,
              "shadowColor": "black",
              "shadowOffset": {
                "height": 12,
                "width": 4,
              },
              "shadowOpacity": 0.6,
              "width": "40%",
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Value
          </Text>
          <Text
            style={
              {
                "color": "black",
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
